flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-oceanbase/src/main/java/org/apache/flink/cdc/connectors/oceanbase/sink/OceanBaseEventSerializationSchema.java

```diff
@@ -0,0 +1,141 @@
+            if (event instanceof CreateTableEvent) {
+                schemaMaps.put(tableId, ((CreateTableEvent) event).getSchema());
+            } else {
+                if (!schemaMaps.containsKey(tableId)) {
+                    throw new RuntimeException("schema of " + tableId + " is not existed.");
+                }
+                schemaMaps.put(
+                        tableId,
+                        SchemaUtils.applySchemaChangeEvent(
+                                schemaMaps.get(tableId), schemaChangeEvent));
+            }
+        }
+        return null;
+    }
+
+    private Record applyDataChangeEvent(DataChangeEvent event) {
+        TableId tableId = event.tableId();
+        Schema schema = schemaMaps.get(tableId);
+        Preconditions.checkNotNull(schema, event.tableId() + " is not existed");
+        Object[] values;
+        OperationType op = event.op();
+        boolean isDelete = false;
+        switch (op) {
+            case INSERT:
+            case UPDATE:
+            case REPLACE:
+                values = serializerRecord(event.after(), schema);
+                break;
+            case DELETE:
+                values = serializerRecord(event.before(), schema);
+                isDelete = true;
+                break;
+            default:
+                throw new UnsupportedOperationException("Unsupported Operation " + op);
+        }
+        return buildDataChangeRecord(tableId, schema, values, isDelete);
+    }
+
+    private DataChangeRecord buildDataChangeRecord(
+            TableId tableId, Schema schema, Object[] values, boolean isDelete) {
+        Preconditions.checkState(
+                Objects.nonNull(tableId.getSchemaName()), "Schema name cannot be null or empty.");
+        com.oceanbase.connector.flink.table.TableId oceanBaseTableId =
+                new com.oceanbase.connector.flink.table.TableId(
+                        tableId.getSchemaName(), tableId.getTableName());
+        TableInfo tableInfo =
+                new TableInfo(
+                        oceanBaseTableId,
+                        schema.primaryKeys(),
+                        schema.getColumnNames(),
+                        Lists.newArrayList(),
+                        null);
+
+        return new DataChangeRecord(
+                tableInfo,
+                isDelete ? DataChangeRecord.Type.DELETE : DataChangeRecord.Type.UPSERT,
+                values);
+    }
+
+    /** serializer RecordData to oceanbase data change record. */
+    public Object[] serializerRecord(RecordData recordData, Schema schema) {
+        List<Column> columns = schema.getColumns();
+        Preconditions.checkState(
+                columns.size() == recordData.getArity(),
+                "Column size does not match the data size");
+        Object[] values = new Object[columns.size()];
+
+        for (int i = 0; i < recordData.getArity(); i++) {
+            OceanBaseRowConvert.SerializationConverter converter =
```

Comment:
I don't think it's necessary. OceanBaseRowConvert.SerializationConverter is a function `@FunctionalInterface`, such as `(index, val) -> val.getByte(index)`, not a data class. Creating it will not have much impact on performance. What do you think?

Reply:
I think the benefit of reusing this SerializationConverter is that we can reduce the judgments of https://github.com/apache/flink-cdc/pull/3360/files#diff-c42ada3283c29232d90a4184ca6196a7f24194229384773ff49792a5d9667318R75

Reply:
Ok，cache added.